src/plugins/auto/auto_executable_network.cpp

```diff
@@ -151,25 +151,30 @@ IE::Parameter AutoExecutableNetwork::GetMetric(const std::string& name) const {
         }
         return decltype(ov::optimal_number_of_infer_requests)::value_type {real};
     } else if (name == ov::execution_devices) {
-        std::lock_guard<std::mutex> lock(_autoSContext->_confMutex);
         std::vector<std::string> exeDevices = {};
-        for (int i = 0; i < CONTEXTNUM; i++) {
-            if (_autoSchedule->_loadContext[i].isEnabled && _autoSchedule->_loadContext[i].isAlready) {
-                if (i == 0 && !_autoSchedule->_loadContext[CPU].executableNetwork._ptr) {
-                    continue;
-                } else {
-                    std::string exeDevices_string = _autoSchedule->_loadContext[i].workName.substr(_autoSchedule->_loadContext[i].workName.find(":") + 1);
-                    if (exeDevices_string == "CPU_HELP")
-                        exeDevices_string = "(CPU)";
-                    std::stringstream ss(exeDevices_string);
-                    std::string item;
-                    while (getline(ss, item, ',')) {
-                        exeDevices.push_back(item);
+        std::string exeDevices_string;
+        if (_autoSContext->_performanceHint == IE::PluginConfigParams::CUMULATIVE_THROUGHPUT) {
+            try {
+                return _autoSchedule->_loadContext[ACTUALDEVICE].executableNetwork->GetMetric(name);
```

Comment:
we should not put this line in return function, the exception may lead to issue

Reply:
Fix it